docs: Updated content on deploying a new account. (#1163)
starknet-io/starknet-docs: components/Starknet/modules/architecture_and_concepts/pages/Accounts/deploying_new_accounts.adoc

[id="deploying_new_accounts"]
= Deploying a new account

You can deploy a new account in the following ways:

* Send a `DEPLOY_ACCOUNT` transaction. This method does not require a preexisting account.

> * Include the `deploy` syscall in the account contract, such as by invoking the Universal Deployer Contract (UDC) with an `INVOKE` transaction. This method requires an existing account to send the `INVOKE` transaction.

Reviewer: ```suggestion
* Using the Universal Deployer Contract (UDC). This method requires an existing account to send the `INVOKE` transaction.
```
Also suggest referencing the UDC documentation

Author: Fixed.

Upon receiving one of these transactions, the sequencer performs the following steps:

> . Verifies that the address has funds to pay for the deployment.

Author: ```suggestion
. Verifies that the address has funds to pay for the deployment.
+
If you use a `DEPLOY_ACCOUNT` transaction, the fees are paid from the address of the deployed account. If you use the UDC, the fees are paid from the sender's account.
```

Author: Fixed.

Reviewer: Remove it from the numbered list and add as a note. For `DEPLOY_ACCOUNT` tx, the fees are charged from the deployed address and from an `INVOKE` tx using the UDC, the fees are charged from the sender's account (and add a link to the transactions types page to invoke tx v1 and v3).

Author: Fixed.

> . Verifies that the address has funds to pay for the deployment.

Author: Fixed.

> . Executes the constructor with the given arguments.

Reviewer: Other way around, first the validate deploy or validate, only then the constructor.

Author: Fixed.

. Runs the respective validation function in the contract, as follows:

> ** When deploying with the `DEPLOY_ACCOUNT` transaction type, the sequencer executes the `+__validate_deploy__+` function in the contract.

Reviewer: ```suggestion
** When deploying with the `DEPLOY_ACCOUNT` transaction type, the sequencer executes the `+__validate_deploy__+` function in the deployed contract.
```

> ** When deploying with the `INVOKE` transaction type and a contract that includes the `deploy` syscall, such as the UDC, the sequencer executes the `+__validate__+` function in the contract.

Reviewer: ```suggestion
** When deploying using the UDC, the sequencer executes the `+__validate__+` function in the contract of the sender address.
```

Author: Fixed.

. Charges fees from the new account address.

> . Sets the account's nonce to `1`.

Reviewer: This is only true in case the account is deployed with the `DEPLOY_ACCOUNT` tx. If by the UDC, the nonce of the deployed account is 0.

Author: Fixed.

> == Deploying a new account with Starkli

Reviewer: I suggest to just keep the link to the Starkli book instead of having this full section.

Author: I got rid of the Prerequisites and the Procedure heading.
